implementations/rust/ockam/ockam_transport_tcp/src/portal/portal_receiver.rs

```diff
@@ -29,8 +31,9 @@ impl Processor for TcpPortalRecvProcessor {
     type Context = Context;
 
     async fn process(&mut self, ctx: &mut Context) -> Result<bool> {
-        let mut buf = [0u8; BUFFER_SIZE];
-        let len = match self.rx.read(&mut buf).await {
+        let mut buf = BytesMut::with_capacity(MAX_PAYLOAD_SIZE);
```

Comment:
> Not as currently written, due to the use of `BytesMut::split` functions.

Could you elaborate a bit?

Reply:
Splitting a BytesMut modifies the underlying buffer reducing it's capacity. So if we allocated this up-front, our buffer would get smaller and smaller over time. This can be handled in various ways (`reserve` prior to the read), but is kind of subtle and could lead to more bugs.

Reply:
Ah, right -- it's the `BufMut` trait that doesn't guarantee that. So the concern about IO is not warranted, but the code after this change is still broken as is.

Reply:
Better solution could splitting BytesMut and then reuniting it back. Definitely don't want to deal with that in this PR. Copy should do for now